### ISSUE TITLE
Remove the Rating pop-up?

### ISSUE THREAD

**Issue by bachig26** [whoeevee/EeveeSpotify#29]
will it be possible to remove the pop-up that displays to rate the app?

**Comment by asdfzxcvbn** [comment]
can you send a screenshot of the popup?

**Comment by bachig26** [comment]
> can you send a screenshot of the popup?

i forgot to take the screenshot, but it will appear from time-to-time like this for in-app rating.

![apprating](https://github.com/whoeevee/EeveeSpotify/assets/96393348/9ebd5155-72fc-4495-b636-e9650e49f8f9)

**Comment by asdfzxcvbn** [comment]
will be removed in the next update

**Comment by bachig26** [comment]
> will be removed in the next update

awesome. thanks.

**Comment by Exit24** [comment]
Any luck with this? On the newest release (EeveeSpotify-5.7-8.9.86) I’m still getting a popup to rate the Spotify app. I can’t even rate the app to stop the popup from coming up again in the future, I’m assuming because it’s a sideloaded/tweaked app. 

![IMG_3188](https://github.com/user-attachments/assets/dfc832d2-7c9a-4e7e-9487-e763db24dfc3)

**Comment by yodaluca23** [comment]
> Any luck with this? On the newest release (EeveeSpotify-5.7-8.9.86) I’m still getting a popup to rate the Spotify app. I can’t even rate the app to stop the popup from coming up again in the future, I’m assuming because it’s a sideloaded/tweaked app. 


Yes, that is likely why but also, if an app requests the pop-up and it's already requested twice in that year it's supposed to not show up, but I have definitely had it pop up more than twice this year and also I even have it turned off in settings. I believe Apple allows developer apps to bypass this because you would kind of want to bypass it for testing so it makes sense...

**Comment by ElliotCHEN37** [comment]
> Any luck with this? On the newest release (EeveeSpotify-5.7-8.9.86) I’m still getting a popup to rate the Spotify app. I can’t even rate the app to stop the popup from coming up again in the future, I’m assuming because it’s a sideloaded/tweaked app. 
> 
> 
> 
> ![IMG_3188](https://github.com/user-attachments/assets/dfc832d2-7c9a-4e7e-9487-e763db24dfc3)
> 
> 

They removed it for some reason. 
https://github.com/whoeevee/EeveeSpotify/commit/b710b1156b1f0a82a7ef6c30af8d217a68376a88

**Comment by asdfzxcvbn** [comment]
try with the debs from my fork: https://github.com/asdfzxcvbn/EeveeSpotify/actions/runs/11536752693

if it works it'll be merged into main

**Comment by Exit24** [comment]
> try with the debs from my fork: https://github.com/asdfzxcvbn/EeveeSpotify/actions/runs/11536752693
> 
> if it works it'll be merged into main

Would I use Sideloady to inject the deb file into the current EeveeSpotify, or a non modified decrypted Spotify ipa?

**Comment by ElliotCHEN37** [comment]
> > try with the debs from my fork: https://github.com/asdfzxcvbn/EeveeSpotify/actions/runs/11536752693
> > if it works it'll be merged into main
> 
> Would I use Sideloady to inject the deb file into the current EeveeSpotify, or a non modified decrypted Spotify ipa?

non modified decrypted

**Comment by whoeevee** [comment]
You may see rating popups when using EeveeSpotify even if you have disabled them in Settings, because the app is in development mode for iOS due to being sideloaded, and StoreKit always displays the rating and review request view with the Submit button grayed out.

You can hook `requestReview` methods of `SKStoreReviewController` to do nothing. This feature was included in a single version of EeveeSpotify, but I had to remove it due to issues. For some reason, Orion couldn’t hook these methods when the app is sideloaded, which led to crashes. I believe these popups are infrequent enough. According to Apple guidelines, it's displayed about maximum of three times within a 365-day period, so it’s not a big deal to press "Cancel".

If the deb provided by zx really works and you don’t see the popup, then the issue with Orion is confirmed, and it will likely be merged. You need to inject the deb from actions into the existing EeveeSpotify .ipa, replacing the current EeveeSpotify in it

**Comment by Exit24** [comment]
> try with the debs from my fork: https://github.com/asdfzxcvbn/EeveeSpotify/actions/runs/11536752693
> 
> if it works it'll be merged into main

Ok, I injected com.eevee.spotify_5.7_iphoneos-arm64.deb into a non modified decrypted Spotify ipa. 

I wasn't sure which file to use so I kind of guessed. I tried injecting all 3 at the same time, but I got an error in Sideloady saying duplicate name eeveespotify.bundle, so I'm guessing I can only do 1 at a time. If I did anything wrong please let me know.

I'll give it some time and see if I get a popup to rate the app.

**Comment by ElliotCHEN37** [comment]
> > try with the debs from my fork: https://github.com/asdfzxcvbn/EeveeSpotify/actions/runs/11536752693
> 
> > 
> 
> > if it works it'll be merged into main
> 
> 
> 
> Ok, I injected com.eevee.spotify_5.7_iphoneos-arm64.deb into a non modified decrypted Spotify ipa. 
> 
> 
> 
> I wasn't sure which file to use so I kind of guessed. I tried injecting all 3 at the same time, but I got an error in Sideloady saying duplicate name eeveespotify.bundle, so I'm guessing I can only do 1 at a time. If I did anything wrong please let me know.
> 
> 
> 
> I'll give it some time and see if I get a popup to rate the app. 

Choose one you need and make sure you also injected SwiftProtobuf, Orion, and CydiaSubstrate

**Comment by Exit24** [comment]
> > > try with the debs from my fork: https://github.com/asdfzxcvbn/EeveeSpotify/actions/runs/11536752693
> > 
> > 
> > > 
> > 
> > 
> > > if it works it'll be merged into main
> > 
> > 
> > Ok, I injected com.eevee.spotify_5.7_iphoneos-arm64.deb into a non modified decrypted Spotify ipa.
> > I wasn't sure which file to use so I kind of guessed. I tried injecting all 3 at the same time, but I got an error in Sideloady saying duplicate name eeveespotify.bundle, so I'm guessing I can only do 1 at a time. If I did anything wrong please let me know.
> > I'll give it some time and see if I get a popup to rate the app.
> 
> Choose one you need and make sure you also injected SwiftProtobuf, Orion, and CydiaSubstrate

I have CydiaSubstrate checked in Sideloady, so I have that injected. Where would I get SwiftProtobuf and Orion from? I can't seem to find them as a deb or dylib file.

**Comment by yodaluca23** [comment]
> 
> 
> 
> 
> 
> 
> 
> 
> > > > try with the debs from my fork: https://github.com/asdfzxcvbn/EeveeSpotify/actions/runs/11536752693
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > if it works it'll be merged into main
> 
> > > 
> 
> > > 
> 
> > > Ok, I injected com.eevee.spotify_5.7_iphoneos-arm64.deb into a non modified decrypted Spotify ipa.
> 
> > > I wasn't sure which file to use so I kind of guessed. I tried injecting all 3 at the same time, but I got an error in Sideloady saying duplicate name eeveespotify.bundle, so I'm guessing I can only do 1 at a time. If I did anything wrong please let me know.
> 
> > > I'll give it some time and see if I get a popup to rate the app.
> 
> > 
> 
> > Choose one you need and make sure you also injected SwiftProtobuf, Orion, and CydiaSubstrate
> 
> 
> 
> I have CydiaSubstrate checked in Sideloady, so I have that injected. Where would I get SwiftProtobuf and Orion from? I can't seem to find them as a deb or dylib file.

SwiftProtobuf is available on the release page, Orion can be downloaded from https://repo.chariz.com

**Comment by asdfzxcvbn** [comment]
no reply
